### ISSUE TITLE
Replacing the backbone network of YOLOv8n with mobilenet found that GFLOPS was particularly high

### ISSUE THREAD

**Issue by Charles-reborned** [ultralytics/ultralytics#17358]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I was training it, it almost filled up my graphics RAM， and here are my codes of MobileBottleneck， MobileConv and C2fMobile
`class MobileBottleneck(nn.Module):
    def __init__(self, c1, c2, k=3, s=1, expansion=4, act=nn.ReLU6):
        super().__init__()
        c_ = int(c2 * expansion)
        self.conv = nn.Sequential(
            MobileConv(c1, c_, k, s, act=act),  # Expansion layer
            MobileConv(c_, c_, k, 1, act=act),  # Depthwise separable convolution
            nn.Conv2d(c_, c2, 1, 1, 0, bias=False)  # Pointwise linear projection
        )

        self.shortcut = (
            nn.Conv2d(c1, c2, 1, 1, 0, bias=False) if s == 1 and c1 != c2 else nn.Identity()
        )

    def forward(self, x):
        return self.conv(x) + self.shortcut(x)`
`
class C2fMobile(C2f):
    """C2f module with MobileBottleneck()."""

    def __init__(self, c1, c2, n=1, shortcut=False, g=1, e=0.5, expansion=4, k=3, s=1):
        """Initialize C2fMobile module with MobileBottleneck."""
        super().__init__(c1, c2, n, shortcut, g, e)
        self.c = int(c2 * e)  # hidden channels

        self.m = nn.ModuleList(
            MobileBottleneck(self.c, self.c, k=k, s=s, expansion=expansion) for _ in range(n)
        )

    def forward(self, x):
        """Forward pass through C2fMobile layer."""
        y = list(self.cv1(x).chunk(2, 1)) 
        y.extend(m(y[-1]) for m in self.m)  
        return self.cv2(torch.cat(y, 1)) 

    def forward_split(self, x):
        """Forward pass using split() instead of chunk()."""
        y = list(self.cv1(x).split((self.c, self.c), 1)) 
        y.extend(m(y[-1]) for m in self.m) 
        return self.cv2(torch.cat(y, 1))  
`
`
class MobileConv(nn.Module):
    def __init__(self, c1, c2, k=1, s=1, act=nn.ReLU6):
        super().__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, padding=k//2, bias=False)
        self.bn = nn.BatchNorm2d(c2)
        self.act = act()  # 使用ReLU6激活函数

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))`
s it because of the ReLU6？the activation function was disabled during the first training, but box_loss became 'nan' when running to the sixth epochs
![20241105193727](https://github.com/user-attachments/assets/267257f9-b64e-4b12-bf47-5557dce8e0e7)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Charles-reborned, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

It seems like you're encountering high GFLOPS when replacing the YOLOv8n backbone with MobileNet. If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it further.

For your custom training ❓ Question, please provide more information about your implementation, including dataset examples and training logs. Also, ensure that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your training process.

Join the lively Ultralytics community for more interactive support. For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧. Prefer detailed discussions? Visit [Discourse](https://community.ultralytics.com) or participate in discussions on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest version of the `ultralytics` package with the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). You can update your setup in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be executed in the following environments, which are fully equipped with all necessary dependencies like [CUDA](https://developer.nvidia.com/cuda), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Please note that this is an automated response, and an Ultralytics engineer will assist you soon with a more detailed look into your issue.

**Comment by Charles-reborned** [comment]
When I used the official configuration of Ghost model training, only 5.2GFLOPS

**Comment by Charles-reborned** [comment]
Or maybe it's because of the code problem. Forgive me for being a novice programmer. I just wrote Mobilenet according to the Ghostnet code because I want to move towards lightweight. And I can't find a direct case of MobilenetV3 introducing yolov8 model on the Internet. Here is my yaml file
`backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  - [-1, 1, MobileConv, [128, 3, 2]] # 1-P2/4
  - [-1, 3, C2fMobile, [128, True]]
  - [-1, 1, MobileConv, [256, 3, 2]] # 3-P3/8
  - [-1, 6, C2fMobile, [256, True]]
  - [-1, 1, MobileConv, [512, 3, 2]] # 5-P4/16
  - [-1, 6, C2fMobile, [512, True]]
  - [-1, 1, MobileConv, [1024, 3, 2]] # 7-P5/32
  - [-1, 3, C2fMobile, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9
  - [-1, 1, SEAttention, [ 16 ] ] #10

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, C2fMobile, [512]] # 13

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, C2fMobile, [256]] # 16 (P3/8-small)

  - [-1, 1, MobileConv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]] # cat head P4
  - [-1, 3, C2fMobile, [512]] # 19 (P4/16-medium)

  - [-1, 1, MobileConv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]] # cat head P5
  - [-1, 3, C2fMobile, [1024]] # 22 (P5/32-large)

  - [[16, 19, 22], 1, Detect, [nc]] # Detect(P3, P4, P5)
`

**Comment by Y-T-G** [comment]
Your `C2fMobile` has a lot more parameters than `C2f`.

**Comment by Charles-reborned** [comment]
I knoww,but I was confused as to why it was so high, then I tried to mix mobile and Ghost and found that the effect was not bad

**Comment by Oil3** [comment]
at layer 18: try changing the concat [-1, 12 ]  to **[-1, 13 ] **

that time i did a typo:
![image](https://github.com/user-attachments/assets/20b062cf-b5ea-4c12-b045-35db6120d1e1)

**Comment by Charles-reborned** [comment]
> @Charles-reborned在第 18 层：尝试将 concat [-1, 12 ] 更改为 **[-1, 13 ] **
> 
> 那次我打错了： ![图像](https://private-user-images.githubusercontent.com/22565084/383415518-20b062cf-b5ea-4c12-b045-35db6120d1e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5Nzg5MzMsIm5iZiI6MTczMDk3ODYzMywicGF0aCI6Ii8yMjU2NTA4NC8zODM0MTU1MTgtMjBiMDYyY2YtYjVlYS00YzEyLWIwNDUtMzVkYjYxMjBkMWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA3VDExMjM1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ZTAwNTA1NDUzM2U1ZTQxYWY1ZTliNTgzZDNmZmFlNWYxOTI0NmUwN2RmYzViMjM5NGVjMmJkMjhkMjcxMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cZb-QJlGekv2f-5MiRCrq7kdoS_Y8c6jjzCVhRuzEMI)

thx bro! I forget this
